File: src/contexts/core/kanji/application/delete-one/delete-kanji-command.ts
import type { Command } from "../../../../shared/cqrs/domain/command";

class DeleteKanjiCommand implements Command {

    static readonly TYPE = "tanukanji.core.kanji.delete:1.0.0";

    constructor(
        public readonly ideogram: string
    ) {}

    get type(): string {
        return DeleteKanjiCommand.TYPE;
    }

}

export { DeleteKanjiCommand };

File: libs/kernel/src/kanjis/application/delete_one/delete_kanji_command.rs
use std::any::Any;

use cqrs::domain::command::Command;


pub struct DeleteKanjiCommand {
    pub ideogram: String,
}

impl DeleteKanjiCommand {

    pub const COMMAND_TYPE: &'static str = "DeleteKanjiCommand";

    pub fn new(ideogram: String) -> Self {
        Self { ideogram }
    }
}

impl Command for DeleteKanjiCommand {
    fn command_type(&self) -> String {
        Self::COMMAND_TYPE.to_string()
    }

    fn as_any(&self) -> &dyn Any {
        self
    }
}